### ISSUE TITLE
incorrect Lint/DefEndAlignment behaviour on multiple modifiers

### ISSUE THREAD

**Issue by Scharrels** [bbatsov/rubocop#4285]
The modifier syntax in Ruby 2.1 can be used to attach multiple modifiers to a method definition, as shown in https://bugs.ruby-lang.org/issues/12495:

```
def cached(name)
  # Rewrite method to include a cache
  return name
end
```

We can use such a method as an extra modifier for a method definition:

```
private cached def test(something)
  return something
end
```

Rubocop doesn't recognize multiple modifiers: the Lint/DefEndAlignment cop generates false positives.

--------

## Expected behavior

This double modifier should not produce any violation if Lint/DefEndAlignment is configured to align with start_of_line.

## Actual behavior

The double modifier produces a violation.

## Further analysis

The problem seems to originate from the `modifier_and_def_on_same_line?`, method, that doesn't recognize multiple modifiers:

https://github.com/bbatsov/rubocop/blob/master/lib/rubocop/cop/mixin/on_method_def.rb#L36

I expect that this method should be aware of the possiblity of nested modifiers, but my knowledge of the parser is too slim to hack it in.

## RuboCop version

```
$ rubocop -V
0.48.1 (using Parser 2.4.0.0, running on ruby 2.1.5 i386-linux-gnu)
```
